test/test_bson.py

```diff
@@ -474,6 +474,13 @@ def test_datetime_encode_decode(self):
         dt2 = BSON.encode({"date": dt1}).decode()["date"]
         self.assertEqual(dt1, dt2)
 
+    def test_large_datetime_truncation(self):
+        # Ensure that a large datetime is truncated correctly.
+        dt1 = datetime.datetime(9999, 1, 1, 1, 1, 1, 999999)
+        dt2 = BSON.encode({"date": dt1}).decode()["date"]
+        self.assertEqual(dt2.microsecond, 999000)
+        self.assertEqual(dt2.second, dt1.second)
```

Comment:
I can reproduce this on Windows with 32-bit and 64-bit Python 3.X. So this fix does actually work.